internal/private/private.go

```diff
@@ -98,6 +99,8 @@ type PutBlobOptions struct {
 
 	EmptyLayer bool // True if the blob is an "empty"/"throwaway" layer, and may not necessarily be physically represented.
 	LayerIndex *int // If the blob is a layer, a zero-based index of the layer within the image; nil otherwise.
+
+	TOCDigest digest.Digest // If specified, the blob can be looked up in the destination also by its TOC digest.
```

Comment:
If there is a match, `TryReusingBlob*` should have found it I think.

Reply:
dropped